python/tvm/relay/transform/quantize/_quantizer.py

```diff
@@ -0,0 +1,133 @@
+
+        # num_orig_outputs is -1 if output is not a Tuple, else is length of tuple
+        if isinstance(self.original_func.body, tvm.relay.expr.Tuple):
+            self.num_orig_outputs = len(self.original_func.body)
+        else:
+            self.num_orig_outputs = -1
+
+        # Partition the func into sub functions containing the patterns we want to quantize
+        partitioned_func = self.original_func
+        for q_pattern in self.patterns:
+            partitioned_func = q_pattern.pattern.partition(partitioned_func)
+
+        # Get rid of first and last par
+        partitioned_func = skip_partitions(partitioned_func, skip_first, skip_last)
+        # Add outputs necessary for calibration
+        tuple_subgraph_func = partition_outputs(partitioned_func)
+
+        # Lower partitioned funcs and store in a mod
+        self.tuple_subgraph_func = lower_partitions(tuple_subgraph_func)
+
+        # Rewrite the multi-output graph to be quantized, and lower partitioned funcs
+        outs = rewrite_partitions(self.patterns, tuple_subgraph_func)
+        q_tuple_subgraph_func = outs["new_out"]
+
+        # Information about each partition used for calibration
+        self.partition_infos = outs["infos_"]
+
+        # Lower quantized partitions and store in a mod
+        self.q_tuple_subgraph_func = lower_partitions(q_tuple_subgraph_func)
+
+        # Create a function containing just the quantized original graph
+        quantized_func = self.q_tuple_subgraph_func
+        if self.num_orig_outputs == -1:
+            self.quantized_func = relay.Function(
+                self.q_tuple_subgraph_func.params, quantized_func.body.fields[0]
+            )
+        else:
+            tuple_body = relay.Tuple(quantized_func.body.fields[self.num_orig_outputs])
+            self.quantized_func = relay.Function(self.q_tuple_subgraph_func.params, tuple_body)
+
+
+def prerequisite_optimize(func, params=None):
+    """Prerequisite optimization passes for quantization. Perform "DynamicToStatic",
+    "SimplifyInference", "FoldConstant", "FoldScaleAxis" before quantization.
+
+    Parameters
+    ---------
+    params : dict of str to NDArray
+        Parameters to use during calibration.
+
+    Returns
+    -------
+    preopt_func : relay.Function
+        The original function with optimizations needed before quantization applied.
+    """
+    optimize = tvm.transform.Sequential(
+        [
+            relay.transform.DynamicToStatic(),
+            relay.transform.SimplifyInference(),
+            relay.transform.FoldConstant(),
+            relay.transform.FoldScaleAxis(),
+        ]
+    )
+
+    if params is not None:
+        func = relay.build_module.bind_params_by_name(func, params)
+
+    mod = tvm.ir.IRModule.from_expr(func)
+    # AlterOpLayout inserts pads and other ops in weird places, so we disable it
+    with relay.build_config(opt_level=3, disabled_pass=["AlterOpLayout"]):
```

Comment:
Why do you need to disable this pass if you're calling your own sequential set of passes instead of build?